src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -811,6 +811,68 @@ public void RecordProperties_10()
                 );
         }
 
+        [ConditionalFact(typeof(DesktopOnly), Reason = ConditionalSkipReason.RestrictedTypesNeedDesktop)]
+        [WorkItem(48115, "https://github.com/dotnet/roslyn/issues/48115")]
+        public void RestrictedTypesAndPointerTypes()
+        {
+            var src = @"
+class C<T> { }
+static class C2 { }
+ref struct RefLike{}
+
+unsafe record C( // 1
```

Comment:
Please add similar test(s) for fields and auto-properties explicitly declared in a record.